### PR TITLE
[source-kyve] Release/0.3.0

### PR DESCRIPTION
**Pull request by christopherbrumm**
## What
These changes solve the provided possibility of Incremental Syncs. Besides this, some minor improvements have been made.

## How

- Handling of offset and cursor_value has been improved
- next_page_token is being used correctly
- Gateway URL-bases have been updated
- Adding Tendermint data normalization for large data rows.

## Review guide
1. `source_kyve/stream.py`
2. `source_kyve/spec.yaml`

## 🚨 User Impact 🚨
These changes will enable the support of Incremental Syncs. Besides this, the core schema has been updated accordingly to support more data types.

For Tendermint data sources, a new kind of normalization has been added. Due to large data rows (>100MB), the source was not usable for BigQuery. The Tendermint normalization extracts the nested events out of the data row in order to write it as independent row.

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Running CI to test changes.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Hello @christopherbrumm it looks both unit tests and acceptance tests failed.
Can you check and fix them? The report is [here](https://ffad14a2ec87eec88ac830bf0989d2a07212f6f79b42393356be912-apidata.googleusercontent.com/download/storage/v1/b/airbyte-ci-reports-multi/o/airbyte-ci%2Fconnectors%2Ftest%2Fpull_request%2Frelease_040%2F1717677937%2Fe5c059a2f12e4997705b6845795bed7d863ae658%2Fsource-kyve%2F0.3.0%2Foutput.html?jk=AS1gM8Nvzg3PEcDAVHBWrwbYhNllVyu1nC9uQPunTUSyNsz2glYHtGzF0bGdhM3CuvYXRYROEaPkzUK2KFaw1GUZr0L7o7elpVJD7KcLTL95KM59F1TgFqCXyqGzFWzYaMvn_cYrtB8NHxz72kzyeHMV1ktji9ytM25xqqFw8q3aXP1acGio0dxx9PXrOHAeLav2_DDyPxgMV7DRENC2SbnXCOq3nqqt5O2STu_VwWSFTVj6jR0wZR4sO_HYFOESqaI2Fkvr8DHWvM_Q6pQJZTjYKL6GVAi0iY8GhiHxhGtQ2dca2tTmfecrydUUGm8MkeZ-sSdoHdelQp2y5G3ej2EBouBkGVgIkm2p_iN8lhjJ0KaYOE9SWRmkSdvdBW4SfdQ3gx5jDFX46sYHGJkNHBJYyssE1ZmuwXwQJxJD1ecbf4DlB1THXXeedeMckQA6MIq_4hZE3uaPRBE80mefVz3Et2bm9hFLiERQ94DRvw9__FHIAQUQTG_ab10DhpzgbXeOl9rjtfGGcnBmLA0h7AcNtPGFSZDHlwGjcvxhlNXTLBtCF6RpcH97hJzzHyZNvun1MRQJSsqrzKH0k8COQkigYzb-8YQcWhQwLt_zO5LqV8_YEjnDa7QLhKWaaBSa3sLfBHS5mgZ6_cc_H1IHstqoiEPk_FaqJFgln6KwWBnzuo4gAv_3CD90XT7Ft9TfQVDZ2P9-SERZiA3xK9MyTXPeM31OotFvaxm1ogkVsMB9yme1cdQv0vBapDKZhclogaTQhFBXAvO-tDXVJeg6-H_ghszb7LIRmlqi1ogFQ19UEt1lnBPG26_bpsBYVQgbx2Ia_afVnWbWfiYUJRbIhigdRg415CENeze_V5hfuUYvSaB48HRRuNkQMGUqKVA714gK2NiSdRQIMioOhq1-IuB4Kofb3YH2VCs3YmjODMT4iC69F2CFCjIxTjAgN8mU3R5eDIwHE9d5SbQ4mnWUQpKJx9JFRkb1TTWFqSLoJP3pBFLhRJnnI0XpPEWwwvNpDCjlCP-pSIccuh3cDNT93-3WpeuFpqUjifsJ8LzOXnqvMAbYWLcnl5vbYS0UA-dtVxB8lZVautYZCQF2uRA7Cp5tB9BQXAKOWUayEOqkf7HjLSY-yZLLgW-sm0QcZYL0ce4Tk8hCLj2ovULGgxeXdY_rP3YCaJM5vxDYE1rqV9C0wnwBipiI2TpYpTMB6yV5r59rhT24_acmrhvgzvCNs0wMRP0sWUH8s8XQ04cyV-lVhyGXPBF4EfUYTUUFZJpERMN8EFtH2lVLYYh7rGTp4zQqLhOQJcJV7SBeWpj12nNLywFZ9RjFWmOO6KhaxKPnUcbVf41jmyzgMssicKV3Yv6J_SXnY0yUo8Nl_OZcLqb-Tn4ROSSAkaTrC1cn7O6KymYj&isca=1)

**Comment by christopherbrumm** [comment]
Hey @marcosmarxm, just updated the tests.

File: airbyte-integrations/connectors/source-kyve/metadata.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ data:
   connectorSubtype: api
   connectorType: source
   definitionId: 60a1efcc-c31c-4c63-b508-5b48b6a9f4a6
-  dockerImageTag: 0.2.2
+  dockerImageTag: 0.3.0
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-kyve/metadata.yaml:5]: @christopherbrumm you need to bump the project.toml version too and run poetry lock to update dependencies.

File: airbyte-integrations/connectors/source-kyve/source_kyve/stream.py [hunk 2]
```diff
@@ -55,22 +56,45 @@ def __init__(self, config: Mapping[str, Any], pool_data: Mapping[str, Any], **kw
...
-            "required": ["key", "value"],
-        }
+        if self._tendermint_normalization:
+            # This is KYVE's default schema and won't be changed.
+            schema = {
+                "$schema": "http://json-schema.org/draft-04/schema#",
+                "type": "object",
+                "properties": {
+                    "height": {"type": "integer"},
+                    "value": {"type": "any"},
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-kyve/source_kyve/stream.py:83]: Schema doesn't support type `any`

File: airbyte-integrations/connectors/source-kyve/source_kyve/stream.py [hunk 3]
```diff
@@ -55,22 +56,45 @@ def __init__(self, config: Mapping[str, Any], pool_data: Mapping[str, Any], **kw
...
+                    "offset": {"type": "string"},
+                },
+                "required": ["key", "value"],
+            }
+        else:
+            # This is KYVE's default schema and won't be changed.
+            schema = {
+                "$schema": "http://json-schema.org/draft-04/schema#",
+                "type": "object",
+                "properties": {"key": {"type": "string"}, "value": {"type": "any"}, "offset": {"type": "string"}},
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-kyve/source_kyve/stream.py:95]: Again, or you build a parser to map to the correct value type of cast to string a default.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Please check tests [output](https://ffa05008bf251f8157a5f72f53a8f807fa0e66f0e9608b93e01e1e7-apidata.googleusercontent.com/download/storage/v1/b/airbyte-ci-reports-multi/o/airbyte-ci%2Fconnectors%2Ftest%2Fpull_request%2Frelease_040%2F1717774738%2F8d78a14da413186d8ca15c75c0cb447d77d9747b%2Fsource-kyve%2F0.3.0%2Foutput.html?jk=AS1gM8O8RCE1gEFtkawqllUOxhu_ixU0NeAIOd6tMurCheA8N3gQ_W12jIcHeJitOreXbJSmdJrx8yiwmzXcllt66sGNks7j8EWUz3SkmHHyX7clqC7_PogPG8JaWuh9At3eGgddIziMkZOPhkNpYTVCBCOUtlF3rqnsfEyTCYtW0Jh-URDqFGOl8rdowiMurym04EGzHNlEN3Ig0lvOq7_niavJi93g-aW3OV1RHeWt0xV7xoAfpkSE56nJdeEIVobu9fM3TgY9dpaL4XP68BUX7_aMjcwLRug6sEw6qW_TAMIt3yItLWlQXDt4o1yMNxkRjEpHJRl5dCpAPeXEZuRcxswSZUQuni5t3LZYfjFRmcqzYbzjMHeHoMGZeFovKI6xU6MvWgjdgl_TpRyDnYrsBW7VbiVHqq0ItJVr-XS8eGZQGPd8uv1k0bMM53Teu00ANe81GHmcLkHdttbE219Jzxvuek-AIbTvL-Nkk3h4CKuT1gZmN0k7dihJjMOTEl_wMTS9jw8O-dEcF4CEXo_haSxuM52bBtJS9BfuB_Dk6L3hisqfkemvcr7CvxXIW0WTw8vQloFwQMhC8fm1bLFp-mTKMtp_LaQFyt6woe584Cf6B3soPKs9J5q-BXi3qmNovzMSUSkiTQkU7yUlAi80XMApACg0AdOHC2y3AjKv4corqFuMIof9PoWx4I0rZ1C8nJ-WO86yrI-HADdg1EFX1qWS56IfZsUrhk_c9jlVHbEo1uZTTnMioXF7BI6KEUh5hfz-AhUboiO3M5BRiJXyk38L9pxafllSGIM7GbmnauAH04ND8S9a2j0Oen5wVi7vzNiFlzBQLH87Pru1u79pu3eK3dWmYRg25acPibcILYkB5sux17jrxWLuRLUm0X-H_cmCZuMslKG452C-g4y0iUw2hUazYxskU7ZEFjLn1s1oxSUFBqeewcrWvzq_bdSp9MwnADZ9y8a_-6Qx3RZDjG1xueotcMY20UJZkwlOUV8_pOv-_Lsfr9GRM4xSw3egbo68BpQTkb4Uuf5NZQRu-avl7fMtGfjwIjxZtqHtTQruvModFu7gM1xp8o83Ghxd-f1vq4ycfBYiHIkD3BrrZsJNKp9cSLbug4MXJ8Kh4dny2uulD4soI-wf54n4VCEpaqSzO835AHUMuoVBmwYYpzSMecIuYQXfoBTGJy0ktyyqWxHwnwcPcb9-zPOqfT2T2JMIRdsde2NL3KKIamCXBPB5krChqiiLK_VcRDkc46rrIlbQz3ey3aJYpX966AM1WI1gZnV-BEhySIbL_ACcHl6PK8I310BH6fUBlrVJPYVj8oVjsOwDpgti-MQw3GiE1rmkuqSS1pxG0RM-1NPiUv1qAC-adnYE2mmYc-KUHhjFhs1L6vyl-8I2QASxdYme&isca=1) still some tests are failing, please make sure they're running properly in local dev. Make some comments about QA failures. Integration tests are related to timeout, so maybe there is some functions need to be improved

**Comment by marcosmarxm** [comment]
@christopherbrumm please request my review after you fix the changes.

**Comment by marcosmarxm** [comment]
@christopherbrumm did you have time to take my last comment?

**Comment by christopherbrumm** [comment]
> @christopherbrumm did you have time to take my last comment?

Thanks for the recommendations - implemented everything.

`airbyte-ci` is still taking a lot of time, although the syncs are running smoothly.

**Comment by marcosmarxm** [comment]
@christopherbrumm trigger CI tests let's wait.

**Comment by marcosmarxm** [comment]
@christopherbrumm, can you update your branch with the latest code from the main branch? Also, is it possible to allow maintainers to push changes to your branch/fork?

**Pull request closed without merging**